apps/next/src/pages/Settings/Pages/ChangePassword/hooks/useValidate.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  PasswordValidationResult,
+  validatePasswordStrength,
+} from '@/lib/passwordValidation';
+import { useCallback, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useFormFieldTouched } from './useFormFieldTouched';
+
+export type PasswordForm = Partial<{
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+}>;
+
+const MIN_PASSWORD_LENGTH = 8;
+
+const INITIAL_ERRORS: PasswordForm = {};
+
+export function useValidate() {
+  const { t } = useTranslation();
+  const [errors, setErrors] = useState<PasswordForm>(INITIAL_ERRORS);
+  const [isFormValid, setIsFormValid] = useState(false);
+  const [passwordStrength, setPasswordStrength] = useState<
+    PasswordValidationResult | undefined
+  >(undefined);
+  const { isTouched, getFieldRef } = useFormFieldTouched<keyof PasswordForm>();
+
+  const validateForm = useCallback(
+    (
+      currentPassword: string,
+      newPassword: string,
+      confirmedPassword: string,
+    ) => {
+      let hasError = false;
+      const newErrors = new Proxy<PasswordForm>(
+        {},
+        {
+          set(target, prop, value) {
+            target[prop] = value;
+            hasError = true;
+            return true;
+          },
```

Comment:
Setters usually return the value that was set, not necessarily `true` -- JustSaying™

<img width="108" height="119" alt="Image" src="https://github.com/user-attachments/assets/27313085-2714-42b1-b2b1-b54e43a696e3" />

```suggestion
          set(target, prop, value) {
            hasError = true;
            return target[prop] = value;
          },
```